components/api/src/main/java/com/hotels/styx/api/ClientCookieDecoder.java

```diff
@@ -0,0 +1,350 @@
+/*
+  Copyright (C) 2013-2020 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.api;
+
+import io.netty.handler.codec.DateFormatter;
+import io.netty.handler.codec.http.cookie.ClientCookieEncoder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.CharBuffer;
+import java.util.Date;
+
+import static com.hotels.styx.api.CookieUtil.firstInvalidCookieNameOctet;
+import static com.hotels.styx.api.CookieUtil.firstInvalidCookieValueOctet;
+import static com.hotels.styx.api.CookieUtil.unwrapValue;
+import static io.netty.util.internal.ObjectUtil.checkNotNull;
+
+/**
+ * A <a href="http://tools.ietf.org/html/rfc6265">RFC6265</a> compliant cookie decoder to be used client side.
+ * <p>
+ * It will store the way the raw value was wrapped in {@link NettyCookie#setWrap(boolean)} so it can be
+ * eventually sent back to the Origin server as is.
+ *
+ * @see ClientCookieEncoder
+ */
+public final class ClientCookieDecoder {
```

Comment:
This class needs to be package private, ideally.

Reply:
All external references removed and it's package private now (so we are using Netty's original ClientCookieDecoder in some tests).